src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -378,11 +384,29 @@ private async Task<IConfidentialClientApplication> BuildConfidentialClientApplic
             }
             catch (Exception ex)
             {
-                _logger.LogInformation(ex, "Exception acquiring token for a confidential client. ");
+                _logger.LogInformation(
+                    ex,
+                    string.Format(CultureInfo.InvariantCulture, 
+                    "Exception acquiring token for a confidential client. "));
                 throw;
             }
         }
 
+        internal void CheckApplicationOptionsContainClientSecret(ConfidentialClientApplicationOptions applicationOptions)
+        {
+            if (string.IsNullOrEmpty(applicationOptions.ClientSecret))
+            {
+                string msg = string.Format(CultureInfo.InvariantCulture, "Client secret cannot be null or whitespace, " +
+                    "and must be included in the configuration of the web app when calling a web API. " +
```

Comment:
A certificate can be used instead of a secret.

Reply:
@bgavrilMS : yes, and we have an enhancements issue for this: (https://github.com/AzureAD/microsoft-identity-web/issues/12).

For the moment, proposing an incremental approach